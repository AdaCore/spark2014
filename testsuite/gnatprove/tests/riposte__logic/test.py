from test_support import *

prove_all (["--counter-example=on"])
