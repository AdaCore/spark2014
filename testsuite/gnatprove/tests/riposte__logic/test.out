logic.adb:17:22: info: assertion proved (CVC4: 2 VC)
logic.adb:17:33: warning: unreachable branch (CVC4: 1 VC)
logic.adb:18:22: info: assertion proved (CVC4: 2 VC)
logic.adb:18:33: warning: unreachable branch (CVC4: 1 VC)
logic.adb:19:22: info: assertion proved (CVC4: 1 VC)
logic.adb:20:22: info: assertion proved (CVC4: 2 VC)
logic.adb:24:22: medium: assertion might fail, cannot prove not (if F then F else true)
logic.adb:24:37: warning: unreachable branch (CVC4: 1 VC)
logic.adb:26:30: info: assertion proved (CVC4: 1 VC)
logic.adb:27:22: medium: assertion might fail, cannot prove not (if F then T else true) (e.g. when F = False and T = True)
logic.adb:27:37: warning: unreachable branch (CVC4: 1 VC)
logic.adb:29:30: info: assertion proved (CVC4: 1 VC)
logic.adb:30:33: medium: assertion might fail, cannot prove F (e.g. when F = False and T = True)
logic.adb:32:30: info: assertion proved (CVC4: 1 VC)
logic.adb:33:22: medium: assertion might fail, cannot prove not (if T then T else true)
logic.adb:43:22: info: assertion proved
logic.adb:44:22: info: assertion proved (CVC4: 1 VC)
logic.adb:45:22: info: assertion proved (CVC4: 1 VC)
logic.adb:46:22: info: assertion proved
logic.adb:50:30: info: assertion proved (CVC4: 1 VC)
logic.adb:51:22: medium: assertion might fail, cannot prove F = T (e.g. when F = False and T = True)
logic.adb:53:30: info: assertion proved (CVC4: 1 VC)
logic.adb:54:22: medium: assertion might fail, cannot prove T = F (e.g. when F = False and T = True)
logic.adb:59:04: warning: analyzing unreferenced procedure "De_Morgan"
logic.adb:62:30: info: assertion proved (CVC4: 1 VC)
logic.adb:63:22: info: assertion proved (CVC4: 3 VC)
logic.adb:65:30: info: assertion proved (CVC4: 1 VC)
logic.adb:66:22: info: assertion proved (CVC4: 2 VC)
logic.adb:70:30: info: assertion proved (CVC4: 1 VC)
logic.adb:71:22: medium: assertion might fail, cannot prove (not (A and B)) = (not A and not B) (e.g. when A = True and B = False) [possible explanation: subprogram at line 59 should mention A and B in a precondition]
logic.adb:73:30: info: assertion proved (CVC4: 1 VC)
logic.adb:74:22: medium: assertion might fail, cannot prove (A or B) = (not A and not B) (e.g. when A = True and B = False) [possible explanation: subprogram at line 59 should mention A and B in a precondition]
logic.adb:76:30: info: assertion proved (CVC4: 1 VC)
logic.adb:77:22: medium: assertion might fail, cannot prove not (A or B) = (A and not B) (e.g. when A = False and B = False) [possible explanation: subprogram at line 59 should mention A and B in a precondition]
logic.adb:79:30: info: assertion proved (CVC4: 1 VC)
logic.adb:80:22: medium: assertion might fail, cannot prove not (A or B) = (not A and B) (e.g. when A = False and B = False) [possible explanation: subprogram at line 59 should mention A and B in a precondition]
logic.adb:83:04: warning: analyzing unreferenced procedure "Contraposition"
logic.adb:86:30: info: assertion proved (CVC4: 1 VC)
logic.adb:87:22: info: assertion proved (CVC4: 2 VC)
logic.adb:92:19: info: postcondition proved (CVC4: 3 VC)
logic.adb:105:14: info: initialization of "R" proved
logic.adb:108:04: warning: analyzing unreferenced procedure "XOR_Test_B"
logic.adb:111:30: info: assertion proved (CVC4: 1 VC)
logic.adb:112:22: info: assertion proved (CVC4: 5 VC)
logic.adb:114:30: info: assertion proved (CVC4: 1 VC)
logic.adb:115:22: info: assertion proved (CVC4: 4 VC)
logic.adb:121:22: info: postcondition proved (CVC4: 2 VC)
logic.adb:128:19: info: postcondition proved (CVC4: 4 VC)
logic.adb:135:04: warning: analyzing unreferenced procedure "Implies_Test_1"
logic.adb:138:22: medium: assertion might fail (e.g. when A = False and B = False and C = False) [possible explanation: subprogram at line 135 should mention A and B and C in a precondition]
